scripts/sanity_chk/sanitylib.py

```diff
@@ -1958,6 +1958,27 @@ def setup_handler(self):
             instance.handler.generator_cmd = self.generator_cmd
             instance.handler.generator = self.generator
 
+    def verify_if_results_filled(self):
+        """
+        Verifies if results for all test cases in the instance exist.
+        It can happen that some (or all) test cases are missing in
+        the instance.results dictionary (due to the instance being filtered,
+        test code not following sanitycheck's rules etc). It is required for
+        the full report to be filled correctly.
+        """
+        if self.instance.status == 'passed':
+            for case in self.instance.testcase.cases:
+                if case not in self.instance.results:
+                    self.instance.results[case] = 'PASS'
+        if self.instance.status == 'failed':
+            for case in self.instance.testcase.cases:
+                if case not in self.instance.results:
+                    self.instance.results[case] = 'FAIL'
+        if self.instance.status == 'skipped':
+            for case in self.instance.testcase.cases:
+                if case not in self.instance.results:
+                    self.instance.results[case] = 'SKIP'
+
```

Comment:
why not just change like this
`

        for case in self.instance.testcase.cases:

            if case not in self.instance.results:

                self.instance.results[case] =self.instance.status.upper
`